src/box/lua/config/applier/credentials.lua

```diff
@@ -819,49 +889,35 @@ local wait_sync
 -- {type = obj_type, name = obj_name} - perform a per-object sync, skip
 --                                      return message
 local function sync_credentials_worker()
-    while true do
-        local obj_to_sync = sync_tasks:get()
+    fiber.self():name('sync_credentials', { truncate = true })
 
-        if obj_to_sync.type == 'BLOCKING_FULL_SYNC' or
-                obj_to_sync.type == 'BACKGROUND_FULL_SYNC' then
-
-            if box.info.ro then
-                log.verbose('credentials: the database is in the read-only ' ..
-                    'mode. Waiting for the read-write mode to set up the ' ..
-                    'credentials given in the configuration. If another ' ..
-                    'instance in the replicaset will write these new ' ..
-                    'credentials, this instance will receive them and will ' ..
-                    'skip further actions.')
-                box.ctl.wait_rw()
-            end
+    if not schema_is_upgraded then
+        local not_upgraded_alert_key = 'not_upgraded_schema'
 
-            -- On config reload, drop the old list of registered objects.
-            target_object_map = {
-                ['role'] = {},
-                ['space'] = {},
-                ['function'] = {},
-                ['sequence'] = {},
-            }
+        config._aboard:set({
+            type = 'warn',
+            message = 'credentials: the schema has an old version ' ..
+                      'and user/roles can not be loaded. Consider executing ' ..
+                      'box.schema.upgrade() to perform an upgrade.'
+        }, {key = not_upgraded_alert_key})
```

Comment:
Nits regarding the message:

* I would say 'database schema', because just 'schema' seems too common. We have YAML configuration schema, for example.
* I wouldn't say 'loaded', because actually they can't be written into the database. 'Loading' sounds more like when something is read.
* 'user/roles' -> 'users/roles'?
* Should we say 'users/roles/privileges'? Technically speaking a privilege is a separate piece of data stored in the `_priv` system space. If a reader knows it, just 'users/roles' may not include privileges for he/she.